common/src/main/java/com/microsoft/identity/common/internal/ui/webview/AzureActiveDirectoryWebViewClient.java

```diff
@@ -165,6 +181,26 @@ private boolean handleUrl(final WebView view, final String url) {
         } else if (isRedirectUrl(formattedURL)) {
             Logger.info(TAG, "Navigation starts with the redirect uri.");
             processRedirectUrl(view, url);
+        } else if (ssoNonce != null && mSsoTokenHandler != null) {
```

Comment:
This will prevent it from triggering on random other requests.  But it would be better to know the "original url" before we got redirected, since it should "match" (for some definition of match).